pytorch_lightning/callbacks/pt_callbacks.py

```diff
@@ -124,9 +124,10 @@ def on_epoch_end(self, epoch, logs=None):
         if current is None:
             warnings.warn(
                 f'Early stopping conditioned on metric `{self.monitor}`'
-                f' which is not available. Available metrics are: {",".join(list(logs.keys()))}',
+                f' which is not available, so early stopping will not work.'
+                f' Available metrics are: {",".join(list(logs.keys()))}',
                 RuntimeWarning)
-            stop_training = True
+
```

Comment:
then you should return `True`

Reply:
Not exactly. Return `True` was before and it caused the interruption of the training if the required metric was not found. And now it just gives a warning and training just proceeds as though without early stopping. The point is that the callback should not stop the training if it can't find the metrics. 

Actually, in the current implementation this branch is not reachable because we check for the availability of the metric in the trainer initialization. But my idea was that if we decide to set early_stopping to True by default, then it can be used to give a warning but not to stop the training.

You can also look at #524 for better understanding.